src/components/SelectionList/BaseSelectionList.tsx

```diff
@@ -666,7 +667,7 @@ function BaseSelectionList<TItem extends ListItem>(
         // reseting the currrent page to 1 when the user types something
         setCurrentPage(1);
 
-        updateAndScrollToFocusedIndex(newSelectedIndex);
+        updateAndScrollToFocusedIndex(shouldResetFocusedIndex ? 0 : newSelectedIndex);
```

Comment:
@waterim This function will not be executed because an early return in line 657

`(prevTextInputValue === textInputValue && flattenedSections.selectedOptions.length === prevSelectedOptionsLength)` will return true

Reply:
Oh, yes, for the correct testing yes, missed that as I was able only to test changing from debit to credit and from credit to debit